### PR TITLE
fix: [CDS-78819]: Adding Env variable AWS_STS_REGIONAL_ENDPOINTS=regional for EKS kube config behind FF CDS_EKS_ADD_REGIONAL_PARAM

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Describe your changes
Adding Env variable AWS_STS_REGIONAL_ENDPOINTS=regional for EKS kube config behind FF CDS_EKS_ADD_REGIONAL_PARAM

### kube config Env Variable after the changes
Screenshot 2023-09-13 at 11 14 59 AM

### After toggling off the feature flag
Screenshot 2023-09-13 at 11 56 55 AM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52775)

### CODE REVIEW

**Comment by namantalaycha10** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/eks/AwsEKSV2DelegateTaskHelper.java [hunk 1]
```diff
@@ -124,6 +130,12 @@ private Exec getEksExecConfig(String clusterName, AwsInternalConfig awsInternalC
       List<EnvVariable> envVariables = new ArrayList<>();
       envVariables.add(awsAccessKey);
       envVariables.add(awsSecretAccessKey);
+      if (isAddRegionalParam) {
+        EnvVariable awsRegionalEndpoint =
+            EnvVariable.builder().name(AWS_STS_REGIONAL_ENDPOINTS).value(AWS_STS_REGIONAL).build();
+        envVariables.add(awsRegionalEndpoint);
+      }
```
Comment (yogesh-chauhan) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/eks/AwsEKSV2DelegateTaskHelper.java:137]: we should add this always if the FF is true.
That means, it should be outside this `if` block

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java [hunk 2]
```diff
@@ -888,7 +888,10 @@ public enum FeatureName {
   CDS_IMPROVED_HELM_DEPLOYMENT_TRACKING(
       "Feature Flag enables improved tracking for Helm deployments that use workloads label selectors to track deployed pods",
       HarnessTeam.CDP),
-  SEI_ENABLED("FF to enable SEI for harness platform users", HarnessTeam.SEI);
+  SEI_ENABLED("FF to enable SEI for harness platform users", HarnessTeam.SEI),
+  CDS_EKS_ADD_REGIONAL_PARAM(
+      "This flag adds the AWS_STS_REGIONAL_ENDPOINTS=regional environment variable while creating the kubenetes config",
```
Comment (yogesh-chauhan) [inline 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:893]: Lets mention that this happens for EKS infrastructure

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java (hunk 2 above)
Comment (lovish1999) [reply 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:893]: :nit: kubernetes ?

**Comment by namantalaycha10** [comment]
trigger uts

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java (hunk 2 above)
Comment (namantalaycha10) [reply 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java]: done

**Comment by namantalaycha10** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 8 of 13 files at r1, 5 of 6 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52775)* status: 14 file(s) to review. 1 of 2 approvals obtained.

**Comment by namantalaycha10** [comment]
trigger messagecheck

**Comment by namantalaycha10** [comment]
trigger messagecheck

**Review by lovish1999 — APPROVED** [review]

**Comment by namantalaycha10** [comment]
trigger review

**Pull request merged**